apollo-router/src/spec/query/transform.rs

```diff
@@ -158,6 +292,27 @@ pub(crate) fn field(
     else {
         return Ok(None);
     };
+
+    for argument in def.arguments.iter() {
+        if let Some(var) = argument.value.as_variable() {
+            visitor
+                .state()
+                .used_variables
+                .insert(var.as_str().to_string());
+        }
+    }
```

Comment:
Values can have variables deeply nested, and there can be multiple variables per argument.
```
query Test($a: String, $b: String) {
  foo(arg: { someField: [[$a, "bar", $b]] })
}
```

It's works for arguments in both fields and directives